sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/rx/HybridSearchQueryTest.java

```diff
@@ -0,0 +1,220 @@
+import static com.azure.cosmos.rx.TestSuiteBase.safeClose;
+import static com.azure.cosmos.rx.TestSuiteBase.safeDeleteDatabase;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Ignore("TODO: Ignore these test cases until the public emulator is released.")
+public class HybridSearchQueryTest {
+    protected static final int TIMEOUT = 30000;
+    protected static final int SETUP_TIMEOUT = 80000;
+    protected static final int SHUTDOWN_TIMEOUT = 20000;
+
+    protected static Logger logger = LoggerFactory.getLogger(HybridSearchQueryTest.class);
+
+    private final String containerId = UUID.randomUUID().toString();
+    private CosmosAsyncClient client;
+    private CosmosAsyncDatabase database;
+    private CosmosAsyncContainer container;
+
+    @BeforeClass(groups = {"query", "split"}, timeOut = SETUP_TIMEOUT* 10)
+    public void before_HybridSearchQueryTest() {
+        //set up the client
+        client = new CosmosClientBuilder()
+            .endpoint(TestConfigurations.HOST)
+            .key(TestConfigurations.MASTER_KEY)
+            .buildAsyncClient();
+
+        database = createDatabase(client, UUID.randomUUID().toString());
+
+        PartitionKeyDefinition partitionKeyDef = new PartitionKeyDefinition();
+        ArrayList<String> paths = new ArrayList<String>();
+        paths.add("/id");
+        partitionKeyDef.setPaths(paths);
+
+        CosmosContainerProperties containerProperties = new CosmosContainerProperties(containerId, partitionKeyDef);
+        containerProperties.setIndexingPolicy(populateIndexingPolicy());
+        containerProperties.setFullTextPolicy(populateFullTextPolicy());
+        database.createContainer(containerProperties, ThroughputProperties.createManualThroughput(10000)).block();
+        container = database.getContainer(containerId);
+
+        List<Document> documents = loadProductsFromJson();
+        for (Document doc : documents) {
+            container.createItem(doc).block();
+        }
+
+        // Wait for 10 minutes to allow indexing
+        try {
+            Thread.sleep(10 * 60 * 1000); // 8 minutes in milliseconds
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Interrupted while waiting for indexing", e);
+        }
+    }
+
+    @AfterClass(groups = {"query"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        safeDeleteDatabase(database);
+        safeClose(client);
+    }
+
+    @Test(groups = {"query"}, timeOut = TIMEOUT)
+    public void hybridQueryTest() {
+
+        String query = "SELECT TOP 10 c.id, c.text, c.title FROM c WHERE FullTextContains(c.text, 'John') OR " +
+            "FullTextContains(c.title, 'John') ORDER BY RANK FullTextScore(c.title, ['John'])";
+        List<Document> resultDocs = container.queryItems(query, new CosmosQueryRequestOptions(), Document.class).byPage()
```

Comment:
please also share one diagnostics example

Reply:
I tried running printing diagnostics but somehow the diagnostics are being added. I know the place where the code is missing for that. I'll create a work item for that and tag it to this PR. But do we need to block this PR for that?

Reply:
no - just tracking it and following up with a PR to fix is good I think

Reply:
Created a follow up issue: https://github.com/Azure/azure-sdk-for-java/issues/43008